Document Alpha 3535.0.0 changes for /etc overlay and flatcar-reset (#279)
flatcar/flatcar-docs: docs/provisioning/ignition/boot-process.md

### Reprovisioning

To trigger a new Ignition run, you should use the `flatcar-reset` tool (available from Alpha 3535.0.0) for a (selective) cleanup of the root filesystem during the next boot. It takes care of cleaning up old state (e.g., files from the old configuration or any side effects such as state files) while only keeping the data you want to keep through the `--keep-paths` argument. The paths to keep can be specified as regular expressions. The machine ID can be kept through the `--keep-machine-id` argument (turning it into a kernel cmdline parameter because `/etc/machine-id` can't be preserved directly for systemd first boot semantics). It is also possible to specify that a local or a particular remote Ignition configuration should be used.

When paths to keep are specified, only needed paths should be used and not those set up by the old Ignition config or side effects of it, to really discard the old configuration state. When a path specified is a folder, the contents are preserved as well because `MYPATH/.*` is automatically appended as additonal regular expression for paths to keep.

> To delete the contents of a folder but keep the folder itself, specify it as equivalent regular expression in the form of `'^/etc/mypath'`, `'/etc/mypath$'`, `'/etc/mypat[h]'`, `'/etc/(mypath)'`, or `'(/etc/mypath)'`.

Reviewer: ```suggestion
To delete the contents of a directory but keep the directory itself, specify it as an equivalent regular expression in the form of `'^/etc/mypath'`, `'/etc/mypath$'`, `'/etc/mypat[h]'`, `'/etc/(mypath)'`, or `'(/etc/mypath)'`.
```

> The used regular expression language is that of `egrep`. Assuming you specified `/etc/mypath`, you can test which paths will be deleted with (note the `-not`):

Reviewer: ```suggestion
The used regular expression language is that of `egrep`. Assuming you specified `/etc/mypath`, you can test which paths will be deleted with the following command (note the `-not`):
```

```sh
find / /etc -xdev -regextype egrep -not -regex '(/etc/mypath|/etc/mypath/.*)'
```

> You can tests which path will be kept with with (note the absence of `-not`):

Reviewer: ```suggestion
You can test which path will be kept with the following command (note the absence of `-not`):
```

```sh
find / /etc -xdev -regextype egrep -regex '(/etc/mypath|/etc/mypath/.*)'
```

> Both `/` and `/etc` have to be specified because `/etc` is an overlay mount.

Reviewer: ```suggestion
Both `/` and `/etc` need to be specified because `/etc` is an overlay mount.
```

Meaningful examples are:

- `'/etc/ssh/ssh_host_.*'` to preserve SSH host keys
- `/var/log` to preserve system logs
- `/var/lib/docker` and `/var/lib/containerd` to preserve container state and images

> Example for selectively resetting the OS with retriggering Ignition while keeping SSH host keys, logs, and machine ID:

Reviewer: ```suggestion
An example for selectively resetting the OS with retriggering Ignition while keeping SSH host keys, logs, and machine ID:
```

```sh
sudo flatcar-reset --keep-machine-id --keep-paths '/etc/ssh/ssh_host_.*' /var/log
sudo systemctl reboot
```

#### Technical Details for Manual Ignition Re-runs

> Not recommended but possible is to either manually set `flatcar.first_boot=1` as temporary kernel command line parameter in GRUB or to create the flag file `touch /boot/flatcar/first_boot` (or `/boot/coreos/first_boot` if the machine was updated from CoreOS CL).

Reviewer: ```suggestion
Not recommended but possible is to either manually set `flatcar.first_boot=1` as temporary kernel command line parameter in GRUB or to create the flag file with `touch /boot/flatcar/first_boot` (or `/boot/coreos/first_boot` if the machine was updated from CoreOS CL).
```
